### PR TITLE
Add TSFE (v10)

### PR DESCRIPTION
**Pull request by sypets**
Cherry-picked commit from "main". Add changes for v10.4.

Disclaimer: I am not 100% sure about some of the request attributes. It looks like frontend.controller is available for v11, but not for v10. I could not find this in the changelogs.

### CODE REVIEW

**Comment by DavidBruchmann** [comment]
@sypets The FrontendController is quite old, so I'm not sure if I understand your comment entirely.

**Comment by sypets** [comment]
> @sypets The FrontendController is quite old, so I'm not sure if I understand your comment entirely.

What do you mean?

I meant this:

`$frontendController = $request->getAttribute('frontend.controller'); `

I removed that (and some of the other sections) from the 10.4 part. It would be cool if someone could double check.

These pages are available for v11, but not v10:

* request attributes: https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ApiOverview/RequestLifeCycle/RequestAttributes/Index.html
* subpage: request attribute frontend.controller: https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ApiOverview/RequestLifeCycle/RequestAttributes/FrontendController.html#typo3-request-attribute-frontend-controller

I am having difficulty finding some things in the changelogs, so I am not entirely sure when it was introduced.

**Comment by brotkrueml** [comment]
Yes, I added the request attributes to the docs some months ago, I didn't care about v10 docs (as I expected to much manual work). To test manually if the attribute is available, just add a breakpoint to some frontend class (e.g. controller) and check the `$GLOBALS['TYPO3_REQUEST']->getAttribute('frontend.controller')` is available. I assume, it is. So, as there is no section, don't like it, just mention the attribute.

**Comment by sypets** [comment]
@brotkrueml Thanks for the clarification. In that case I should undo the last commit and start over. Will take a bit, am at work currently. If someone wants to finish this, you are welcome to do so.

**Comment by brotkrueml** [comment]
Just checked: `frontend.controller` request attribute is not available in v10.

**Comment by sypets** [comment]
@brotkrueml 

> Just checked: frontend.controller request attribute is not available in v10.

ok, then the removal was probably fine

File: Documentation/ApiOverview/TSFE/Index.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    rid of this in the future.
+
+TSFE is not available in all contexts. In particular, it is
+only available in frontend contexts, not in the backend or CLI.
+
+Initializing :php:`$GLOBALS['TSFE']` in the backend is sometimes done in code
+examples found online. This is not recommended. TSFE is not initialized in the
+backend context by the Core (and there is usually no need to do this).
+
+From the PHP documentation:
```
Comment (brotkrueml) [inline Documentation/ApiOverview/TSFE/Index.rst:61]: I assume, this can be removed, as TYPO3 v10 is only PHP until 7.4.

Comment (sypets) [reply Documentation/ApiOverview/TSFE/Index.rst]: done

**Review by brotkrueml — APPROVED** [review]

**Comment by brotkrueml** [comment]
@sypets Thanks for your work on this :-)

**Pull request merged**